pyuvdata/uvdata/mir_parser.py

```diff
@@ -2825,7 +3442,50 @@ def _read_compass_solns(self, filename):
 
         return compass_soln_dict
 
-    def _apply_compass_solns(self, compass_soln_dict, apply_flags=True, apply_bp=True):
+    def read_compass_solns(self, filename=None, load_flags=True, load_bandpass=True):
+        """
+        Read in COMPASS-formatted bandpass and flagging solutions.
+
+        Reads in an HDF5 file containing the COMPASS-derived flags and gains tables.
+        These solutions are applied as the data are read in (when calling `load_data`).
+
+        Parameters
+        ----------
+        filename : str
+            Name of the file containing the COMPASS flags and gains solutions.
+        load_flags : bool
+            If set to True, the COMPASS flags will be read into the MirParser object.
+            Default is True.
+        load_gains : bool
+            If set to True, the COMPASS gains will be read into the MirParser object.
+            Default is True.
+
+        Raises
+        ------
+        UserWarning
+            If the COMPASS solutions do not appear to overlap in time with that in
+            the MirParser object.
+        """
+        if not load_flags and not load_bandpass:
+            # Insert snarky no-op comment here
```

Comment:
Reminder to add snarky comment

Reply:
Added in 97f7c9f6c0c8405b61c0a77902b405abfc50738f.